packages/wonder-blocks-data/src/hooks/use-data-internal.js

```diff
@@ -0,0 +1,113 @@
+    // initial value for the result state).
+    const maybeTrack = useContext(TrackerContext);
+    if (result == null && Server.isServerSide()) {
+        maybeTrack?.(handler, options);
+    }
+
+    // We need to update our request when the handler changes or the key
+    // to the options change, so we keep track of those.
+    // However, even if we are hydrating from cache, we still need to make the
+    // request at least once, so we do not initialize these references.
+    const handlerRef = useRef();
+    const keyRef = useRef();
+
+    // This effect will ensure that we fulfill the request as desired.
+    useEffect(() => {
+        // If we are server-side, then just skip the effect. We track requests
+        // during SSR and fulfill them outside of the React render cycle.
+        // NOTE: This shouldn't happen since effects would not run on the server
+        // but let's be defensive - I think it makes the code clearer.
+        if (Server.isServerSide()) {
+            return;
+        }
+
+        // If the handler hasn't changed and the key that the options represent
+        // hasn't changed, then we don't need to make our request.
+        if (
+            handler === handlerRef.current &&
+            handler.getKey(options) === keyRef.current
+        ) {
+            return;
+        }
+
+        // Update our refs to the current handler and key.
+        handlerRef.current = handler;
+        keyRef.current = handler.getKey(options);
+
+        // If we're not hydrating a result, we want to make sure we set our
+        // result to null so that we're in the loading state.
+        if (cachedResult == null) {
+            // Mark ourselves as loading.
+            setResult(null);
+        }
+
+        // We aren't server-side, so let's make the request.
+        // The request handler is in control of whether that request actually
+        // happens or not.
+        let cancel = false;
+        RequestFulfillment.Default.fulfill(handler, options)
+            .then((updateEntry) => {
+                if (cancel) {
+                    return;
+                }
+                setResult(updateEntry);
+                return;
+            })
+            .catch((e) => {
+                if (cancel) {
+                    return;
+                }
+                /**
+                 * We should never get here as errors in fulfillment are part
+                 * of the `then`, but if we do.
+                 */
+                // eslint-disable-next-line no-console
+                console.error(
+                    `Unexpected error occurred during data fulfillment: ${e}`,
+                );
+                setResult({
+                    data: null,
+                    error: typeof e === "string" ? e : e.message,
+                });
+                return;
+            });
+
+        return () => {
+            cancel = true;
+        };
+    }, [handler, options, handlerRef, keyRef, cachedResult]);
```

Comment:
The linter is really annoying then as it demands they be included.

I think the linter for deps is really leading to bad code and would be better to be disabled permanently.

Reply:
I missed `RequestFulfillment.Default.fulfill(handler, options)`.  I think disabling the lint rule in this situation makes sense because we have a custom condition of when things should be re-run.  We could probably work around the linter by storing `options` in a `ref`, but I'd that seems like the tail wagging the dog.